components/reddit/sources/new-link-on-a-subreddit/new-link-on-a-subreddit.js

```diff
@@ -0,0 +1,88 @@
+const reddit = require("../../reddit.app.js");
+const get = require("lodash.get");
+
+module.exports = {
+  key: "new-link-on-a-subreddit",
+  name: "New Link on a subreddit",
+  description: "Emits an event each time a new link is added to a subreddit",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    reddit,
+    subreddit: {
+      type: "string",
+      label: "Subreddit",
+      description: "The subreddit you'd like to watch for new links.",
+      default: "redditdev",
+    },    
+    timer: {
+      label: "Polling schedule",
+      description:
+        "Pipedream polls Reddit for new links on this schedule.",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 10, // by default, run every 10 minute.
+      },
+    },
+    db: "$.service.db",
+  },
+  hooks: {
+    async deploy() {
+      // Emits 10 events on the first run during deploy.
+      let after = null;
+
+      const reddit_things = await this.reddit.getNewSubredditLinks(
+        null,
+        this.subreddit,
+        10
+      );
+      
+      if (get(reddit_things, "data.after") !== undefined) {
+        after = reddit_things.data.after;
+        if (reddit_things.data.children.length > 0) {
+          reddit_things.data.children.forEach((reddit_link) => {
+            console.log(reddit_link.data.name);
+            this.emitRedditEvent(reddit_link);
+          });
+        }
+      }
+
+      this.db.set("after", after);
+    },
+  },
+  methods: {
+    emitRedditEvent(reddit_link) {
+      this.$emit(reddit_link, {
+        id: reddit_link.data.name,
+        summary: reddit_link.data.name,
```

Comment:
I missed this earlier, but you'll want the event summary to be human-readable, so we prefer titles to IDs. I'd change to `reddit_link.data.title`. Technically you can cut down just a bit of code by destructuring and naming the destructured args, like so:

```javascript
emitRedditEvent(event) {
  const { name: id, title: summary } = event.data
  this.$emit(event, {
    id,
    summary,
  })
}
```

Reply:
Ok noted. good to know, i started to use the title as indicated.